File: backend/team6/src/test/java/programmers/team6/mock/VacationRequestsReaderFake.java
package programmers.team6.mock;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import programmers.team6.domain.admin.dto.request.VacationStatisticsRequest;
import programmers.team6.domain.admin.support.VacationRequests;
import programmers.team6.domain.admin.support.VacationRequestsReader;
import programmers.team6.domain.vacation.entity.VacationRequest;

public class VacationRequestsReaderFake extends VacationRequestsReader {

	private final List<VacationRequest> vacationRequests;

	public VacationRequestsReaderFake(VacationRequest... vacationRequests) {
		super(null);
		this.vacationRequests = new ArrayList<>(Arrays.asList(vacationRequests));
	}

	@Override
	public VacationRequests vacationRequestFrom(List<Long> ids, VacationStatisticsRequest request) {
		return new VacationRequests(vacationRequests);
	}
}

File: backend/team6/src/test/java/programmers/team6/mock/VacationRequestsReaderFake.kt
package programmers.team6.mock

import io.mockk.mockk
import programmers.team6.domain.admin.dto.request.VacationStatisticsRequest
import programmers.team6.domain.admin.support.VacationRequests
import programmers.team6.domain.admin.support.VacationRequestsReader
import programmers.team6.domain.vacation.entity.VacationRequest
import programmers.team6.domain.vacation.repository.VacationRequestRepository
import java.util.*

class VacationRequestsReaderFake(vararg vacationRequests: VacationRequest) : VacationRequestsReader(mockk<VacationRequestRepository>()) {
    private val vacationRequests: List<VacationRequest> =
        ArrayList(Arrays.asList(*vacationRequests))

    override fun vacationRequestFrom(ids: List<Long>, request: VacationStatisticsRequest): VacationRequests {
        return VacationRequests(vacationRequests)
    }
}
